### PR TITLE
Format, lint, git hook and update CI

### PR DESCRIPTION
**Pull request by Thykof**
TODO:
- [x] fix some linting errors
- [x] add lint staged
- [x] do not use yarn in commands
- [x] improve the use of prettier for assembly script
- [x] improve the linter used for assembly script

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,61 +1,64 @@
...
+    "private": false,
+    "publishConfig": {
+        "access": "public",
+        "registry": "https://registry.npmjs.org/"
+    },
+    "scripts": {
+        "test": "asp --verbose",
+        "test:ci": "asp --summary",
+        "fmt": "prettier-eslint --write '{*,**/*}.{js,ts,jsx,tsx,json}'",
+        "lint:fix": "eslint --fix --ext .ts . && yarn fmt",
```
Comment (peterjah) [inline package.json:28]: why yarn is use here ? :/

Comment (peterjah) [reply package.json:28]: just use "npm run fmt"

Comment (Thykof) [reply package.json:28]: good catch

File: package.json [hunk 2]
```diff
@@ -24,24 +24,29 @@
...
+    "@as-pect/cli": "^7.0.7",
+    "@assemblyscript/loader": "^0.21.2",
+    "@massalabs/as": "1.0.0",
     "@typescript-eslint/eslint-plugin": "^5.29.0",
     "@typescript-eslint/parser": "^5.29.0",
-    "@as-pect/cli": "^7.0.7",
     "as-console": "^6.0.2",
     "assemblyscript": "^0.21.2",
-    "@assemblyscript/loader": "^0.21.2",
+    "assemblyscript-prettier": "git://github.com/Thykof/assemblyscript-prettier#update-assemblyscript-21",
```
Comment (peterjah) [inline package.json:41]: I found a betterway to handle the dependency version conflict:
leave ```"assemblyscript-prettier": "^1.0.5",```
just add this in your  package.json
    ```"overrides": {
      "assemblyscript": "^0.21.7"
    }```

**Review by peterjah — APPROVED** [review]

**Review by Trottoir — APPROVED** [review]

**Pull request merged**